src/components/Sidebar/SidebarIcon.tsx

```diff
@@ -11,38 +11,39 @@ export interface ISidebarIconProps {
   children: ReactNode;
 }
 
-const SidebarIcon = ({
-  name,
-  isActive,
-  isSidebarExpanded,
-  onClick,
-  children,
-}: ISidebarIconProps) => {
-  return (
-    <button
-      type="button"
-      onClick={() => onClick()}
-      className={`sidebar-background-color-transition flex flex-row align-middle w-full rounded-lg
+const SidebarIcon = forwardRef<HTMLButtonElement, ISidebarIconProps>(
+  function SidebarIcon(props, ref) {
```

Comment:
Was there a reason that you chose to keep `DefaultTooltip` outside of the `SidebarIcon` component and instead forwarded the ref in?

Reply:
@

> Was there a reason that you chose to keep `DefaultTooltip` outside of the `SidebarIcon` component and instead forwarded the ref in?

@ajmcquilkin it won't show anything if I don't use forward ref or other HTML element as Sidebar Icon is a custom component.

Reply:
@ajmcquilkin I know this syntax.

Reply:
Sorry, I should have phrased that better. Instead of writing:

```typescript
<SidebarTab title="Overview" isSidebarExpanded={isSidebarExpanded}>
  <DefaultTooltip text="Messaging">
    <SidebarIcon
      name="Messaging"
      isActive={location.pathname === AppRoutes.MESSAGING}
      isSidebarExpanded={isSidebarExpanded}
      onClick={() => navigateTo(AppRoutes.MESSAGING)}
    >
      <MessagesSquare strokeWidth={1.5} className="w-6 h-6" />
    </SidebarIcon>
  </DefaultTooltip>
</SidebarTab>
```

Is there a reason you didn't define `SidebarIcon` as:

```typescript
<DefaultTooltip ... >
  <button ... >
     ... contents of SidebarIcon ...
  </button>
</SidebarIcon>
```

I think it would be fine forcing all instances of `SidebarIcon` to have a tooltip, and this would make disabling the tooltip easier with the `isActive` prop

Reply:
> Sorry, I should have phrased that better. Instead of writing:
> 
> ```ts
> <SidebarTab title="Overview" isSidebarExpanded={isSidebarExpanded}>
>   <DefaultTooltip text="Messaging">
>     <SidebarIcon
>       name="Messaging"
>       isActive={location.pathname === AppRoutes.MESSAGING}
>       isSidebarExpanded={isSidebarExpanded}
>       onClick={() => navigateTo(AppRoutes.MESSAGING)}
>     >
>       <MessagesSquare strokeWidth={1.5} className="w-6 h-6" />
>     </SidebarIcon>
>   </DefaultTooltip>
> </SidebarTab>
> ```
> 
> Is there a reason you didn't define `SidebarIcon` as:
> 
> ```ts
> <DefaultTooltip ... >
>   <button ... >
>      ... contents of SidebarIcon ...
>   </button>
> </SidebarIcon>
> ```
> 
> I think it would be fine forcing all instances of `SidebarIcon` to have a tooltip, and this would make disabling the tooltip easier with the `isActive` prop

@ajmcquilkin you are like that?

<DefaultTooltip text="Messaging">
              <button>
              <SidebarIcon
                name="Messaging"
                isActive={location.pathname === AppRoutes.MESSAGING}
                isSidebarExpanded={isSidebarExpanded}
                onClick={() => navigateTo(AppRoutes.MESSAGING)}
              >
                <MessagesSquare strokeWidth={1.5} className="w-6 h-6" />
              </SidebarIcon>
              </button>
            </DefaultTooltip>

Reply:
Sorry, not sure I'm understanding your comment. My question is just out of curiosity, doesn't seem to make a functionality difference.